### PR TITLE
Alpha Animal update

### PR DESCRIPTION
**Pull request by Breadbox2k19**
You know what to do and so do I.

!!!Important!!!!
Compatibility between Alpha Animals and Vanilla Faction Expanded: Mechanoids are now patched!!!
VFE Alpha Mechanoids now works with CE, However, it will only work IF Combat Extended if loaded after both Alpha Animals and Vanilla Faction Expanded: Mechanoids!!!!!!!!!!!

## Testing

Check tests you have performed:
- [x] Compiles without warnings
- [x] Game runs without errors
- [x] (For compatibility patches) ...with and without patched mod loaded
- [x] Playtested a colony (specify how long)

### CODE REVIEW

**Comment by CyanSam** [comment]
Maybe while you're at it patching the gigantelope extra carry weight heddif would be a good idea. Thanks for the patches tho.

**Comment by Breadbox2k19** [comment]
> Maybe while you're at it, patching the Gigantelope extra carry weight heddif would be a good idea. The numbers seem balanced enough so I think just changing the CarryingCapacity to CarryWeight could do the job. Thanks for the patches tho.

So you are saying that gig antelope carries too much stuff? Have to say, it's really overpowered, maybe I'll patch it together with all of the other stuff.

**Comment by CyanSam** [comment]
> > Maybe while you're at it, patching the Gigantelope extra carry weight heddif would be a good idea. The numbers seem balanced enough so I think just changing the CarryingCapacity to CarryWeight could do the job. Thanks for the patches tho.
> 
> So you are saying that gig antelope carries too much stuff? Have to say, it's really overpowered, maybe I'll patch it together with all of the other stuff.

No not that it's overpowered, the heddif increases carry capacity in it's vanilla form instead of what I assume should be carry weight which determines how much stuff can be carried in CE.

File: Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_Black_Hive.xml [hunk 1]
```diff
@@ -8,6 +8,36 @@
...
+		</nomatch>
+	</li>
+
+		<li Class="PatchOperationAdd">
+			<xpath>/Defs/ThingDef[defName="AA_BlackSpider" or defName="AA_BlackSpelopede"]/comps</xpath>
+			<value>
+				<li>
+				  <compClass>CombatExtended.CompPawnGizmo</compClass>
+				</li>
+				<li Class="CombatExtended.CompProperties_Suppressable" />
```
Comment (N7Huntsman) [inline Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_Black_Hive.xml:37]: Why would these creatures be suppressable? Base insects aren't.

File: Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_Black_Hive.xml [hunk 2]
```diff
@@ -105,6 +135,7 @@
 					<MeleeDodgeChance>0.08</MeleeDodgeChance>
 					<MeleeCritChance>0.18</MeleeCritChance>
 					<MeleeParryChance>0.15</MeleeParryChance>
+          					<Suppressability>0.25</Suppressability>
```
Comment (N7Huntsman) [inline Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_Black_Hive.xml:138]: See the above.

File: Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_Black_Hive.xml [hunk 3]
```diff
@@ -188,6 +230,7 @@
 					<MeleeDodgeChance>0.09</MeleeDodgeChance>
 					<MeleeCritChance>0.45</MeleeCritChance>
 					<MeleeParryChance>0.25</MeleeParryChance>
+          					<Suppressability>0.2</Suppressability>
```
Comment (N7Huntsman) [inline Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_Black_Hive.xml:233]: See the above.

**Review by N7Huntsman — CHANGES_REQUESTED** [review]
Most of the balance changes look fine. Still need to test and check balance on the new creatures, but just one small concern in the meantime.

File: Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_Black_Hive.xml (hunk 1 above)
Comment (Breadbox2k19) [reply Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_Black_Hive.xml:37]: Yes, that two blackhive insects are range combatants, I want add some surpressibility to them to make fire superiority a bit more useful.

**Comment by Breadbox2k19** [comment]
> > > Maybe while you're at it, patching the Gigantelope extra carry weight heddif would be a good idea. The numbers seem balanced enough so I think just changing the CarryingCapacity to CarryWeight could do the job. Thanks for the patches tho.
> > 
> > 
> > So you are saying that gig antelope carries too much stuff? Have to say, it's really overpowered, maybe I'll patch it together with all of the other stuff.
> 
> No not that it's overpowered, the heddif increases carry capacity in it's vanilla form instead of what I assume should be carry weight which determines how much stuff can be carried in CE.

CE determines carry weight the same way as it is in vanilla.

**Review by N7Huntsman — CHANGES_REQUESTED** [review]

File: Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_Black_Hive.xml (hunk 1 above)
Comment (N7Huntsman) [reply Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_Black_Hive.xml:37]: Hmm. I'll see how it playtests, but I'm not certain of how well it's going to work. I'd suspect they might move to cover but may not be tall enough to effectively fight from cover, depending upon the size of the bugs.

File: Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_DarkBeast.xml [hunk 4]
```diff
@@ -0,0 +1,68 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<Patch>
+	<Operation Class="PatchOperationFindMod">
+		<mods>
+			<li>Alpha Animals</li>
+		</mods>
+		<match Class="PatchOperationSequence">
+			<operations>
+				<li Class="PatchOperationAddModExtension">
+					<xpath>/Defs/ThingDef[defName="AA_Darkbeast"]</xpath>
```
[the remark this answers is not in the document]
Comment (Breadbox2k19) [reply Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_DarkBeast.xml:10]: Forgot to add projectile thingsclass, should be fixed now.

File: Patches/Alpha Animals/VFE_Mechs/ADVPhalanx.xml [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+			</li>
+		
+			
+			<!-- Make it spawn with ammo -->
+			
+			<li Class="PatchOperationAdd">
+				<xpath>/Defs/ThingDef[defName="VFE_Mech_Advanced_Phalanx"]/comps</xpath>
+				<value>
+					<li>
+						<compClass>CombatExtended.CompPawnGizmo</compClass>
```
[the remark this answers is not in the document]
Comment (Breadbox2k19) [reply Patches/Alpha Animals/VFE_Mechs/ADVPhalanx.xml:83]: I don't think adding them breaks anything, if you feel they are just bloat, do feel free to delete them.

File: Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_Black_Hive.xml (hunk 1 above)
Comment (Breadbox2k19) [reply Patches/Alpha Animals/ThingDefs_Races/AlphaAnimals_CE_Patch_Race_Black_Hive.xml:37]: They already seek cover by default, making them suppressible will not change  their behaviour.

**Review by N7Huntsman — APPROVED** [review]
Checked stats to to confirm all animal had melee stats and that their ranged attacks were working. General remarks for future adjustments from testing:
- Erin ranged attack has a very long warm up.
- Mantrap projectile is unusually fast.

VFE Mech Addon Changes:
- Goliaths should not spawn with needleguns, since CE changes how they work.
- The siege breaker weapon is pretty deadly, with heavy damage and excellent accuracy. It may be appropriate to give it a larger still minimum range.

**Pull request merged**